qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -228,34 +260,52 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
             if self._sample_from_iterations:
                 power = np.random.randint(power)
             # Run a grover experiment for a given power of the Grover operator.
-            if self._quantum_instance.is_statevector:
-                qc = self.construct_circuit(amplification_problem, power, measurement=False)
-                circuit_results = self._quantum_instance.execute(qc).get_statevector()
+            if self._sampler:
+                qc = self.construct_circuit(amplification_problem, power, measurement=True)
+                job = self._sampler.run([qc])
+
+                # combine the results
+                try:
+                    results = job.result()
+                except Exception as exc:
+                    raise AlgorithmError("Sampler job failed.") from exc
+
                 num_bits = len(amplification_problem.objective_qubits)
+                circuit_results = {
+                    np.binary_repr(k, num_bits): v for k, v in results.quasi_dists[0].items()
+                }
+                top_measurement = max(circuit_results.items(), key=lambda x: x[1])[0]
+                max_probability = max(circuit_results.items(), key=lambda x: x[1])[1]
```

Comment:
Looking at the grover result it has private_fields for circuit_results and shots. While this logic here creates circuit_results from the sampler, as in done in the old quantum instance code paths too, we do not exposed that as public access in the result. Is this something we should be looking at doing of circuits results is something useful for the result.

I see shots is never set into the result, even in the prior quantum instance paths, so the private field there in the result seems pointless.